moveit_core/robot_state/include/moveit/robot_state/robot_state.h

```diff
@@ -1311,27 +1298,43 @@ as the new values that correspond to the group */
   /** \brief Update the state after setting a particular link to the input global transform pose.*/
   void updateStateWithLinkAt(const LinkModel* link, const Eigen::Isometry3d& transform, bool backward = false);
 
+  /** \brief Look-up the cached transform from the model frame (root of model) to |link_name|.
+   *   Checks if there are any dirty (non-updated) transforms first and updates transforms cache as needed
+   *   A related, more comprehensive function is |getFrameTransform| which includes this functionality
+   *   but also other frame sources
+   */
   const Eigen::Isometry3d& getGlobalLinkTransform(const std::string& link_name)
   {
     return getGlobalLinkTransform(robot_model_->getLinkModel(link_name));
   }
 
-  const Eigen::Isometry3d& getGlobalLinkTransform(const LinkModel* link)
-  {
-    updateLinkTransforms();
-    return global_link_transforms_[link->getLinkIndex()];
-  }
-
+  /** \brief Look-up the cached transform from the model frame (root of model) to |link_name|.
+   *   Checks if there are any dirty (non-updated) transforms first and updates transforms cache as needed
+   *   A related, more comprehensive function is |getFrameTransform| which includes this functionality
+   *   but also other frame sources
+   */
+  const Eigen::Isometry3d& getGlobalLinkTransform(const LinkModel* link);
+
+  /** \brief Look-up the cached transform from the model frame (root of model) to |link_name|.
+   *   Checks if there are any dirty (non-updated) transforms first and updates transforms cache as needed
+   *   A related, more comprehensive function is |getFrameTransform| which includes this functionality
+   *   but also other frame sources
+   *   As opposed to the visual links in |getGlobalLinkTransform|, this function returns
+   *   the collision link transform used for collision checking.
+   */
   const Eigen::Isometry3d& getCollisionBodyTransform(const std::string& link_name, std::size_t index)
   {
     return getCollisionBodyTransform(robot_model_->getLinkModel(link_name), index);
   }
 
-  const Eigen::Isometry3d& getCollisionBodyTransform(const LinkModel* link, std::size_t index)
-  {
-    updateCollisionBodyTransforms();
-    return global_collision_body_transforms_[link->getFirstCollisionBodyTransformIndex() + index];
-  }
+  /** \brief Look-up the cached transform from the model frame (root of model) to |link_name|.
```

Comment:
|link_name| should be |link|

The |index| parameter is undocumented

Reply:
i believe its:
> specify which collision body to lookup, if more than one exists